src/test/java/com/exasol/bucketfs/uploadnecassity/ChecksumUploadNecessityCheckStrategyTest.java

```diff
@@ -0,0 +1,139 @@
+package com.exasol.bucketfs.uploadnecassity;
+
+import static com.exasol.bucketfs.BucketConstants.DEFAULT_BUCKET;
+import static com.exasol.bucketfs.BucketConstants.DEFAULT_BUCKETFS;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.startsWith;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.io.TempDir;
+
+import com.exasol.bucketfs.*;
+
+//[itest->dsn~conditional-upload~1]
+class ChecksumUploadNecessityCheckStrategyTest extends AbstractBucketIT {
+    private static Connection connection;
+    private Bucket bucket;
+    private ChecksumUploadNecessityCheckStrategy uploadCheck;
+
+    @BeforeAll
+    static void beforeAll() throws SQLException {
+        connection = EXASOL.createConnection();
+    }
+
+    @AfterAll
+    static void afterAll() throws SQLException {
+        connection.close();
+    }
+
+    private SyncAwareBucket getBucket() {
+        final var bucketConfiguration = getDefaultBucketConfiguration();
+        return SyncAwareBucket.builder()//
+                .ipAddress(getContainerIpAddress()) //
+                .httpPort(getMappedDefaultBucketFsPort()) //
+                .serviceName(DEFAULT_BUCKETFS) //
+                .name(DEFAULT_BUCKET) //
+                .readPassword(bucketConfiguration.getReadPassword()) //
+                .writePassword(bucketConfiguration.getWritePassword()) //
+                .monitor(createBucketMonitor()) //
+                .build();
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        this.bucket = getBucket();
```

Comment:
Why is a new bucket allocated with each test if the creation is not parameterized?

Reply:
it's non final. --> Tests modify it. I want to have a clear object for each test

Reply:
Accepted.